lib/puma/server.rb

```diff
@@ -217,6 +217,7 @@ def run_lopez_mode(background=true)
     def handle_servers_lopez_mode
       begin
         check = @check
+        handle_check
```

Comment:
These calls to `handle_check` (and the change to make `handle_check` non-blocking) works around a bug triggered by the `TestIntegrationSingle#test_term_suppress` test, where the TERM signal is sent to the process before the server is fully booted up, causing it to enter an endless loop because the calls to `IO.select` raise an error (because the IO is already closed), so the 'stop server' command is never received.